src/openforms/prefill/contrib/haalcentraal/plugin.py

```diff
@@ -152,15 +153,29 @@ def get_co_sign_values(
         )
 
     def check_config(self):
+        config = HaalCentraalConfig.get_solo()
+        assert isinstance(config, HaalCentraalConfig)
+        if not config.service:
+            raise InvalidPluginConfiguration(_("Service not selected"))
+
+        client = config.service.build_client()
+
         try:
-            config = HaalCentraalConfig.get_solo()
-            assert isinstance(config, HaalCentraalConfig)
-            if not config.service:
-                raise InvalidPluginConfiguration(_("Service not selected"))
-
-            client = config.service.build_client()
-            # FIXME: haal centraal v2 client does not support GET methods
-            client.retrieve("test", "test")
+            if config.version == HaalCentraalVersion.haalcentraal13:
+                client.retrieve("ingeschrevenpersonen", "999990676")
```

Comment:
this really should try to retrieve a resource that doesn't exist with an invalid "BSN".

The problem if you start using "real requests" is that those result in calls to the national registration, which may incur a cost but at least is frowned upon because they are unnecessary calls and there's no legal basis to request the details of a "random" person.

So the intent here is that you check the connection to the service - if you get a 404 that means it's working because the service replied with something useful (rather than a 403, 5XX... which would imply issues in the configuration).